autogpt_platform/frontend/src/app/(platform)/auth/callback/route.ts

```diff
@@ -11,56 +11,12 @@ async function shouldShowOnboarding() {
   );
 }
 
-// Default redirect path - matches the home page redirect destination
-const DEFAULT_REDIRECT_PATH = "/marketplace";
-
-// Validate redirect URL to prevent open redirect attacks and malformed URLs
-function validateRedirectUrl(url: string): string {
-  try {
-    const cleanUrl = url.trim();
-
-    // Check for completely invalid patterns that suggest URL corruption
-    if (
-      cleanUrl.includes(",") || // Any comma suggests concatenated URLs
-      cleanUrl.includes(" ") // Spaces in URLs are problematic
-    ) {
-      console.warn(
-        "Detected corrupted redirect URL (likely race condition):",
-        cleanUrl,
-      );
-      return DEFAULT_REDIRECT_PATH;
-    }
-
-    // Only allow relative URLs that start with /
-    if (!cleanUrl.startsWith("/") || cleanUrl.startsWith("//")) {
-      console.warn("Invalid redirect URL format:", cleanUrl);
-      return DEFAULT_REDIRECT_PATH;
-    }
-
-    // Additional safety checks
-    if (cleanUrl.split("/").length > 5) {
-      // Reasonable path depth limit
-      console.warn("Suspiciously deep redirect URL:", cleanUrl);
-      return DEFAULT_REDIRECT_PATH;
-    }
-
-    // For now, allow any valid relative path (can be restricted later if needed)
-    return cleanUrl;
-  } catch (error) {
-    console.error("Error validating redirect URL:", error);
-    return DEFAULT_REDIRECT_PATH;
-  }
-}
-
 // Handle the callback to complete the user session login
 export async function GET(request: Request) {
   const { searchParams, origin } = new URL(request.url);
   const code = searchParams.get("code");
 
-  // if "next" is in param, use it as the redirect URL
-  const nextParam = searchParams.get("next") ?? "/";
-  // Validate redirect URL to prevent open redirect attacks
-  let next = validateRedirectUrl(nextParam);
+  let next = "/marketplace";
```

Comment:
We don't have a user generated `next` param, not sure why this logic was here 🤔 

<img width="800" height="238" alt="Image" src="https://github.com/user-attachments/assets/963d04a6-86e6-42c1-9838-7a26ca587667" />

The logic should be simple for now, after login with Google either re-direct to `/onboarding` if they need to complete it or to `/marketplace`